nilearn/connectome/connectivity_matrices.py

```diff
@@ -231,6 +231,77 @@ def sym_to_vec(symmetric, discard_diagonal=False):
     return symmetric[..., tril_mask] / scaling[tril_mask]
 
 
+def vec_to_sym_matrix(vec, diagonal=None):
+    """Return the symmetric matrix given its flattened lower triangular part.
+    Acts on the last dimension of the array if not 1-dimensional.
+    Diagonal can be encompassed in vec or given separately. In both cases, note
+    that diagonal elements are multiplied by sqrt(2).
+
+    Parameters
+    ----------
+    vec : numpy.ndarray, shape (..., p * (p + 1) /2)
+        The input array.
+
+    diagonal : numpy.ndarray, shape (..., p), optional
+        The diagonal array to be stacked to vec.
+
+    Returns
+    -------
+    sym : numpy.ndarray, shape (..., p, p) or (..., p + 1, p + 1) if diagonal
+        is given seperately.
+        The output symmetric array.
+
+    Note
+    ----
+    This function is meant to be the inverse of sym_matrix_to_vec.
+
+    See also
+    --------
+    nilearn.connectome.sym_matrix_to_vec
+    """
+    n = vec.shape[-1]
+    # solve p * (p + 1) / 2 = n subj. to p > 0
```

Comment:
I think they help explaining what's done, because otherwise the value of `p` comes from nowhere.
You disagree ?